agent/utils/version/mysql.go

```diff
@@ -15,43 +15,92 @@
 package version
 
 import (
+	"context"
 	"regexp"
+	"strconv"
 	"strings"
 
 	"gopkg.in/reform.v1"
 )
 
-// regexps to extract version numbers from the `SHOW GLOBAL VARIABLES WHERE Variable_name = 'version'` output.
+// MySQLVendor represents MySQL vendor.
+type MySQLVendor byte
+
+// MySQLVendor represent MySQL vendors.
+const (
+	OracleVendor MySQLVendor = iota + 1
+	PerconaVendor
+	MariaDBVendor
+)
+
+// MySQLVersion represent major, minor numbers of mysql version separated by comma.
+type MySQLVersion struct {
+	text   string
+	number float64
+}
+
+const (
+	perconaComment = "percona"
+	mariaDBComment = "mariadb"
+	debianComment  = "debian"
+
+	versionQuery = `SHOW GLOBAL VARIABLES WHERE Variable_name = 'version'`
+	commentQuery = `SHOW GLOBAL VARIABLES WHERE Variable_name = 'version_comment'`
+)
+
 var (
+	// Regexps to extract version numbers from the `SHOW GLOBAL VARIABLES WHERE Variable_name = 'version'` output.
 	mysqlDBRegexp = regexp.MustCompile(`^\d+\.\d+`)
+	// Vendors to represent MySQLVendor enum in string format with default value unknown.
+	vendors = [...]string{"unknown", "oracle", "percona", "mariadb"}
```

Comment:
If you need string representations than maybe custom type will work better than enum.
https://github.com/percona/pmm/blob/634520f51213c2a13c45201806f181f9b46fd96c/agent/runner/jobs/job.go#L25-L34

Reply:
actually a matter of taste in my opinion, but this approach with enam looks prettier for me, since there are cases like mysql version when you need to specify a range of values and that's where number representation of enum comes in handy.